WebStoreTerehin/Controllers/EmployeesController.cs

```diff
@@ -1,63 +1,148 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Linq;
 using Microsoft.AspNetCore.Mvc;
+using WebStoreTerehin.Controllers.Infrastructure.Interfaces;
+using WebStoreTerehin.Data;
 using WebStoreTerehin.Models;
+using WebStoreTerehin.ViewModels;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
 
 namespace WebStoreTerehin.Controllers
 {
+    //[Route("Users")]
     public class EmployeesController : Controller
     {
-        private static readonly List<Employee> _Employees = new List<Employee>
+        private readonly IEmployeesData _EmployeesData;
+        public EmployeesController(IEmployeesData EmployeesData) { _EmployeesData = EmployeesData; }
+
+        //[Route("All")]
+        public IActionResult Index() => View(_EmployeesData.Get());
+
+        public IActionResult AddEmployees() => View();
+
+        //[Route("User-{id}")]
+        public IActionResult Details(int id)
         {
-            new Employee
-            {
-                Id=1,
-                Surname="Иванов",
-                Name = "Иван",
-                Patronymic="Иванович",
-                Age=39,
-                DateOfEmployment=Convert.ToDateTime("01.01.2010")
-            },
-            new Employee
+            var employee = _EmployeesData.GetById(id);
+            if (employee is null)
+                return NotFound();
+
+            return View(employee);
+        }
+
+        #region Edit
+        [HttpGet]
+        public IActionResult Edit(int? id)
+        {
+            if (id is null) return View(new EmployeesViewModel());
+
+            if (id < 0)
+                return BadRequest();
+
+            var employee = _EmployeesData.GetById((int)id);
+            if (employee is null)
+                return NotFound();
+
+            return View(new EmployeesViewModel
             {
-                Id=2,
-                Surname="Петров",
-                Name = "Петр",
-                Patronymic="Петрович",
-                Age=49,
-                DateOfEmployment=Convert.ToDateTime("01.01.2015")
-            },
-            new Employee
+                Id = employee.Id,
+                FirstName = employee.Name,
+                LastName = employee.Surname,
+                Patronymic=employee.Patronymic,
+                Age = employee.Age,
+                DateOfEmployment=employee.DateOfEmployment
+            }) ;
+        }
+        [HttpPost]
+        public IActionResult Edit(EmployeesViewModel Model)
+        {
+            if (Model is null)
+                throw new ArgumentNullException(nameof(Model));
+
+            //Валидация
+
+            Regex rgx = new Regex(@"^[А-Я]{1}([а-я])+$");
+            if (Model.LastName is null)
+                ModelState.AddModelError("LastName", "Заполните поле Фамилия");
+            else if (!rgx.IsMatch(Model.LastName))
+                ModelState.AddModelError("LastName", "Недопустимые символы в поле Фамилия");
```

Comment:
В данном случае лучше использовать:
```suggestion
                ModelState.AddModelError(nameof(Employee.LastName), "Недопустимые символы в поле Фамилия");
```